### PR TITLE
Support 0 node agent pool and block manual scale CAS enabled pool

### PR DESCRIPTION
**Pull request by gtxistxgao**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your PR is merged into master branch, a new PR will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repo and upgrade the version in the PR but do not modify `src/index.json`.

### CODE REVIEW

File: src/aks-preview/azext_aks_preview/custom.py [hunk 1]
```diff
@@ -1518,10 +1518,11 @@ def aks_scale(cmd,  # pylint: disable=unused-argument
         raise CLIError('There are more than one node pool in the cluster. '
                        'Please specify nodepool name or use az aks nodepool command to scale node pool')
 
-    if node_count == 0:
-        raise CLIError("Can't scale down to 0 nodes.")
     for agent_profile in instance.agent_pool_profiles:
         if agent_profile.name == nodepool_name or (nodepool_name == "" and len(instance.agent_pool_profiles) == 1):
+            if agent_profile.enable_auto_scaling:
+                raise CLIError("Cannot scale cluster autoscaler enabled agent pool.")
```
Comment (marwanad) [inline src/aks-preview/azext_aks_preview/custom.py:1524]: `nodepool` instead of `agent pool` to be consisted with the above error messages?

Comment (gtxistxgao) [reply src/aks-preview/azext_aks_preview/custom.py:1524]: make sense. let me update it.

Comment (gtxistxgao) [reply src/aks-preview/azext_aks_preview/custom.py]: updated

**Comment by yonzhan** [comment]
AKS

**Pull request merged**